exposed-core/src/main/kotlin/org/jetbrains/exposed/sql/vendors/MysqlDialect.kt

```diff
@@ -308,7 +318,10 @@ open class MysqlDialect : VendorDialect(dialectName, MysqlDataTypeProvider, Mysq
 
     override fun isAllowedAsColumnDefault(e: Expression<*>): Boolean {
         if (super.isAllowedAsColumnDefault(e)) return true
-        val acceptableDefaults = arrayOf("CURRENT_TIMESTAMP", "CURRENT_TIMESTAMP()", "NOW()", "CURRENT_TIMESTAMP(6)", "NOW(6)")
+        if ((currentDialect is MariaDBDialect && fullVersion >= "10.2.1") || (currentDialect !is MariaDBDialect && fullVersion >= "8.0.13")) {
```

Comment:
It breaks my logic, that any expression in the versions above `10.2.1` and `8.0.13` for MariaDB and MySql respectively are valid default expressions. It works now because only the check for dates after that check. It could be better to reverse the condition and move call to super at the end of that function. Could be like that:

```
override fun isAllowedAsColumnDefault(e: Expression<*>): Boolean {
    if ((currentDialect is MariaDBDialect && fullVersion < "10.2.1") || (currentDialect !is MariaDBDialect && fullVersion < "8.0.13")) {
        val acceptableDefaults = mutableListOf("CURRENT_TIMESTAMP", "CURRENT_TIMESTAMP()", "NOW()", "CURRENT_TIMESTAMP(6)", "NOW(6)")
        if (e.toString().trim() in acceptableDefaults && isFractionDateTimeSupported()) return true
    }
    return super.isAllowedAsColumnDefault(e)
}
```
With this variant the new test is green.

I just worry to miss cases which should return `true` in new databases, but `super.isAllowedAsColumnDefault(e)` would return `false` for it. If it's a case, my variant is wrong.

One more moment is that specific constant of a version, probably it could be moved to protected field like `VERSION_OF_GREAT_CHANGE_IN_DATETIMES_DEFAULTS` and overridden inside MariaDBDialect. In this case MysqlDialect as a parent class would not be aware of child classes.